packages/salesforcedx-vscode-core/src/commands/forceProjectCreate.ts

```diff
@@ -0,0 +1,160 @@
+import * as glob from 'glob';
+import * as path from 'path';
+import { Observable } from 'rxjs/Observable';
+import * as vscode from 'vscode';
+import { channelService } from '../channels';
+import { nls } from '../messages';
+import { notificationService } from '../notifications';
+import { CancellableStatusBar, taskViewService } from '../statuses';
+import {
+  CancelResponse,
+  CompositeParametersGatherer,
+  ContinueResponse,
+  EmptyPreChecker,
+  ParametersGatherer,
+  PostconditionChecker,
+  SfdxCommandlet,
+  SfdxCommandletExecutor
+} from './commands';
+
+class ForceProjectCreateExecutor extends SfdxCommandletExecutor<{}> {
+  public build(data: ProjectNameAndPath): Command {
+    return new SfdxCommandBuilder()
+      .withDescription(nls.localize('force_project_create_text'))
+      .withArg('force:project:create')
+      .withFlag('--projectname', data.projectName)
+      .withFlag('--outputdir', data.projectUri)
+      .build();
+  }
+
+  public execute(response: ContinueResponse<ProjectNameAndPath>): void {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: response.data.projectUri
+    }).execute(cancellationToken);
+
+    execution.processExitSubject.subscribe(async data => {
+      if (data != undefined && data.toString() === '0') {
+        await vscode.commands.executeCommand(
+          'vscode.openFolder',
+          vscode.Uri.parse(
+            path.join(response.data.projectUri, response.data.projectName)
+          )
+        );
+      }
+    });
+
+    notificationService.reportExecutionError(
+      execution.command.toString(),
+      (execution.stderrSubject as any) as Observable<Error | undefined>
+    );
+    channelService.streamCommandOutput(execution);
+    CancellableStatusBar.show(execution, cancellationTokenSource);
+    taskViewService.addCommandExecution(execution, cancellationTokenSource);
+  }
+}
+
+export type ProjectNameAndPath = ProjectName & ProjectURI;
+
+export interface ProjectURI {
+  projectUri: string;
+}
+
+export interface ProjectName {
+  projectName: string;
+}
+
+export class SelectProjectName implements ParametersGatherer<ProjectName> {
+  public async gather(): Promise<
+    CancelResponse | ContinueResponse<ProjectName>
+    > {
+    const projectNameInputOptions = {
+      prompt: nls.localize('parameter_gatherer_enter_project_name')
+    } as vscode.InputBoxOptions;
+    const projectName = await vscode.window.showInputBox(
+      projectNameInputOptions
+    );
+    return projectName
+      ? { type: 'CONTINUE', data: { projectName } }
+      : { type: 'CANCEL' };
+  }
+}
+
+export class SelectProjectFolder implements ParametersGatherer<ProjectURI> {
+  public async gather(): Promise<
+    CancelResponse | ContinueResponse<ProjectURI>
+    > {
+    const projectUri = await vscode.window.showOpenDialog({
```

Comment:
Opens native file picker with the settings I set. Only letting it select one folder. Hitting cancel will return undefined.